lnwallet/rpcwallet/remote_signer_client_test.go

```diff
@@ -0,0 +1,721 @@
+package rpcwallet
+
+import (
+	"context"
+	"errors"
+	"math"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/signrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/walletrpc"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc/metadata"
+)
+
+var (
+	// ErrStreamCanceled is returned when the mock stream is canceled by the
+	// remote signer client.
+	ErrStreamCanceled = errors.New("stream canceled")
+
+	// ErrMockResponseErr is a mock error that is returned by the mock
+	// signer server.
+	ErrMockResponseErr = errors.New("mock response error")
+
+	// ErrStreamError is returned when the mock stream creation fails.
+	ErrStreamError = errors.New("stream creation error")
+)
+
+// mockStreamFeeder is a mock implementation of SignCoordinatorStreamFeeder.
+type mockStreamFeeder struct {
```

Comment:
I think it might potentially work, and I also started implementing changing these to use `mock.Mock`, but didn't prio that change with this push. If you think this is an high prio, I'll ensure to address it :)